std/range/primitives.d

```diff
@@ -1987,6 +1987,81 @@ ElementType!R moveAt(R, I)(R r, I i) if (isIntegral!I)
 }
 
 /**
+Pops the first element from $(D r) and returns it.
+It combines $(D front) and $(D popFront) in one method.
+
+---
+R r = [1,2,3];
+assert(r.frontPop == 1)
+assert(r == [2,3])
+---
+ */
+auto frontPop(Range)(ref Range r)
+if (isInputRange!Range)
+{
+    auto e = r.front;
+    r.popFront();
+    return e;
+}
+
+@safe unittest
```

Comment:
Examples should come from ddoc-ed unit tests, not in documentation. All you have to do is add `///` to the previous line